sphinx_autodoc_typehints.py

```diff
@@ -129,11 +129,13 @@ def format_annotation(annotation,
             full_name = 'typing.Optional'
             args_format = '\\[:py:data:`{prefix}typing.Union`\\[{{}}]]'.format(prefix=prefix)
             args = tuple(x for x in args if x is not type(None))  # noqa: E721
-    elif full_name == 'typing.Callable' and args and args[0] is not ...:
+    elif full_name in ['typing.Callable', 'collections.abc.Callable'] \
+            and args and args[0] is not ...:
+        arglist = args[0] if isinstance(args[0], list) else args[:-1]
         formatted_args = '\\[\\[' + ', '.join(
             format_annotation(
                 arg, simplify_optional_unions=simplify_optional_unions)
-            for arg in args[:-1]) + ']'
+            for arg in arglist) + ']'
```

Comment:
These changes seem to be unnecessary. The first element of `args` is never a list. I reverted this and your new tests still pass.